### DISCUSSION TITLE
How to include environment variable in a snippet?

### DISCUSSION THREAD

**Discussion by jimcornmell** [L3MON4D3/LuaSnip#475] [Ideas]
Is it possible to include any environment variable in a snippet definition json file eg `$HOME`:

Not sure what the syntax would be `$HOME` or maybe `$ENV['HOME']` or even `$somefunc("HOME")` ?

```json
{
  "A Description": {
    "prefix": "home",
    "body": [
         "$HOME",
         "$ENV['HOME']"
    ],
....
}
```

**Comment by L3MON4D3** [comment]
Not if the snippet is defined in lsp-syntax, you'll have to use lua for that.
But this seems like a nice addition, could you open an issue to track it?
(extending the lsp-variables is already almost possible, see #464, but I think just adding environment variables to "core"-luasnip (eg not via some user-defined function) would be a nicer approach)

**Comment by jimcornmell** [comment]
Thanks for the quick response, I've created #476 for you.
